apps/blockchain/test/blockchain/transaction/receipt_test.exs

```diff
@@ -18,4 +18,226 @@ defmodule Blockchain.Transaction.ReceiptTest do
+                0, 0, 0, 0, 0>>,
+            topics: []
+          }
+        ]
+      )
+      |> Receipt.serialize()
+      |> ExRLP.encode()
+
+    assert expected_result == result
+  end
+
+  test "encodes receipt with multiple similar logs" do
+    encoded =
+      <<47, 105, 125, 103, 30, 154, 228, 238, 36, 164, 60, 75, 13, 126, 21, 241, 203, 75, 166,
+        222, 21, 97, 18, 13, 67, 185, 164, 232, 196, 168, 166, 238>>
+      |> Receipt.new(
+        265_390,
+        [
+          %EVM.LogEntry{
+            address:
+              <<220, 244, 33, 208, 147, 66, 139, 9, 108, 165, 1, 167, 205, 26, 116, 8, 85, 167,
+                151, 111>>,
+            data:
+              <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0>>,
+            topics: []
+          },
+          %EVM.LogEntry{
+            address:
+              <<220, 244, 33, 208, 147, 66, 139, 9, 108, 165, 1, 167, 205, 26, 116, 8, 85, 167,
+                151, 111>>,
+            data:
+              <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0>>,
+            topics: []
+          }
+        ]
+      )
+      |> Receipt.serialize()
+      |> ExRLP.encode()
+
+    expectected_result =
+      <<249, 1, 154, 160, 47, 105, 125, 103, 30, 154, 228, 238, 36, 164, 60, 75, 13, 126, 21, 241,
+        203, 75, 166, 222, 21, 97, 18, 13, 67, 185, 164, 232, 196, 168, 166, 238, 131, 4, 12, 174,
+        185, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 248, 112, 247, 148, 220, 244,
+        33, 208, 147, 66, 139, 9, 108, 165, 1, 167, 205, 26, 116, 8, 85, 167, 151, 111, 192, 160,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 247, 148, 220, 244, 33, 208, 147, 66, 139, 9, 108, 165, 1, 167, 205, 26, 116, 8, 85,
+        167, 151, 111, 192, 160, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>
+
+    assert encoded == expectected_result
+  end
+
+  test "decodes receipt with multiple different logs" do
```

Comment:
Not sure if it would be helpful, but what do you think of adding a test that decodes a bloom filter, then encodes the result, and we check that they are the same? It would in a sense test the same thing these tests are testing, but it also adds the clarity that the decoding/encoding are inverse operations, right? Anyway, just an idea. Love the thorough testing here though. 👍

Reply:
@germsvel I think the first test tests it https://github.com/poanetwork/mana/blob/72f5d2d97edae204a3469618b18082bf55a968c2/apps/blockchain/test/blockchain/transaction/receipt_test.exs#L14